js/ui/main.js

```diff
@@ -241,7 +241,17 @@ function _initializeUI() {
 
     if (sessionMode.isGreeter && screenShield) {
         layoutManager.connect('startup-prepared', function() {
-            screenShield.showDialog();
+            // We can't show the login dialog (which is managed by the
+            // screenshield) until the PaygManager is initializd, since
+            // we need to check whether the machine is PAYG-locked first.
+            if (paygManager.initialized) {
+                screenShield.showDialog();
+            } else {
+                let paygManagerId = paygManager.connect('initialized', () => {
+                    screenShield.showDialog();
+                    paygManager.disconnect(paygManagerId);
```

Comment:
What’s going to happen in the interim, while waiting for this async operation to finish?

Reply:
Nothing. The screenshield is created already way before this async operation is finished, patiently waiting for that to happen so that `showDialog()` can be called.

Reply:
If you mean "visually", then the answer is again "nothing": the login screen is actually constructed and shown from showDialog(), so nothing will be shown until its called, which is equivalent to the login screen (or the unlock dialog) to maybe taking a few extra miliseconds to show up.